dll/win32/shlwapi/autocomp.c

```diff
@@ -0,0 +1,707 @@
+                case DRIVE_NO_ROOT_DIR:
+                case DRIVE_REMOVABLE:
+                case DRIVE_CDROM:
+                    break;
+                case DRIVE_REMOTE:
+                case DRIVE_RAMDISK:
+                case DRIVE_FIXED:
+                    AC_DoDir(pES, sz, bDirOnly);
+                    break;
+            }
+        }
+    }
+}
+
+static void
+AC_DoURLHistory(AC_EnumString *pES)
+{
+    static const LPCWSTR
+    pszTypedURLs = L"Software\\Microsoft\\Internet Explorer\\TypedURLs";
+    HKEY hKey;
+    LONG result;
+    DWORD i, cbValue;
+    WCHAR szName[32], szValue[MAX_PATH + 32];
+
+    result = RegOpenKeyExW(HKEY_CURRENT_USER, pszTypedURLs, 0, KEY_READ, &hKey);
+    if (result == ERROR_SUCCESS)
+    {
+        for (i = 1; i <= 50; ++i)
+        {
+            StringCbPrintfW(szName, sizeof(szName), L"url%lu", i);
+
+            szValue[0] = 0;
+            cbValue = sizeof(szValue);
+            result = RegQueryValueExW(hKey, szName, NULL, NULL, (LPBYTE)szValue, &cbValue);
+            if (result != ERROR_SUCCESS)
+                continue;
+
+            if (UrlIsW(szValue, URLIS_URL))
+            {
+                AC_EnumString_AddString(pES, szValue);
+            }
+        }
+
+        RegCloseKey(hKey);
+    }
+}
+
+static void
+AC_DoURLMRU(AC_EnumString *pES)
+{
+    static const LPCWSTR
+    pszRunMRU = L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\RunMRU";
+    HKEY hKey;
+    LONG result;
+    DWORD i, cbValue;
+    WCHAR szName[2], szMRUList[64], szValue[MAX_PATH + 32];
+    INT cch;
+
+    result = RegOpenKeyExW(HKEY_CURRENT_USER, pszRunMRU, 0, KEY_READ, &hKey);
+    if (result == ERROR_SUCCESS)
+    {
+        szMRUList[0] = 0;
+        cbValue = sizeof(szMRUList);
+        result = RegQueryValueExW(hKey, L"MRUList", NULL, NULL, (LPBYTE)szMRUList, &cbValue);
+        if (result == ERROR_SUCCESS)
+        {
+            for (i = 0; i < 26; ++i)
+            {
+                if (szMRUList[i] == 0)
+                    break;
+
+                szName[0] = szMRUList[i];
+                szName[1] = 0;
+                szValue[0] = 0;
+                cbValue = sizeof(szValue);
+                result = RegQueryValueExW(hKey, szName, NULL, NULL, (LPBYTE)szValue, &cbValue);
+                if (result != ERROR_SUCCESS)
+                    continue;
+
+                cch = wcslen(szValue);
+                if (cch >= 2 && wcscmp(&szValue[cch - 2], L"\\1") == 0)
+                {
+                    szValue[cch - 2] = 0;
+                }
+
+                if (UrlIsW(szValue, URLIS_URL))
+                {
+                    AC_EnumString_AddString(pES, szValue);
+                }
+            }
+        }
+
+        RegCloseKey(hKey);
+    }
+}
+
+static HRESULT STDMETHODCALLTYPE
+AC_EnumString_Reset(IEnumString* This)
+{
+    DWORD attrs;
+    WCHAR szText[MAX_PATH];
+    AC_EnumString *this = (AC_EnumString *)This;
+    DWORD dwSHACF_ = this->m_dwSHACF_;
```

Comment:
No need to keep the underscore in the name of the local variable (it makes the notation too heavy otherwise).
(+ Same comment for the other functions where a similar situation occurs).